mesonbuild/dependencies/dev.py

```diff
@@ -500,18 +509,26 @@ def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.
 
 
 class JNISystemDependency(SystemDependency):
-    def __init__(self, environment: 'Environment', kwargs: T.Dict[str, T.Any]):
-        super().__init__('jni', environment, kwargs)
+    def __init__(self, environment: 'Environment', kwargs: JNISystemDependencyKW):
+        super().__init__('jni', environment, T.cast(T.Dict[str, T.Any], kwargs))
 
         self.feature_since = ('0.62.0', '')
 
         m = self.env.machines[self.for_machine]
+        assert m
 
         if 'java' not in environment.coredata.compilers[self.for_machine]:
             detect_compiler_for(environment, 'java', self.for_machine)
         self.javac = environment.coredata.compilers[self.for_machine]['java']
         self.version = self.javac.version
 
+        modules: T.Set[str] = set(mesonlib.listify(kwargs.get('modules', [])))
```

Comment:
Why does this need to be converted to a set? It does not seem that you are using any of the properties of a set compared to a list.

Reply:
Wanted to deduplicate in the case someone passed `modules: ['jvm', 'jvm']`

Reply:
But you only ever check `if 'jvm' in modules` and so on, so that doesn't care if it appears multiple times.

Reply:
You win :)